Artemis-xyz/dbt
# Labeling: Adding EVM Flipside Labels

### Summary
1. Adding EVM flipside labels so that we can track deposit contracts for the new stablecoin work. I have NOT materialized `dim_flipside_contracts` yet. If @Sunny77D could give me a quick review since I know our labeling system has a cycle in it and I dont want to introduce major issues. 

## Questions
1. There is currently some copy pasta code, it is done on purpose to be clear what chains are able to move forward in the labeling system. Can unify this with a `where in` clause but not sure if we want to keep it explicit

## Test Plan
The plan is to test the new tables to make sure they are what is expected when being passed to downstream models. Once all the `dim_flipside_<chain>_contracts` are what is expected I will then test `dim_flipside_contracts` this is the last model before the new tags are pass into the tagging system. Once `dim_flipside_contracts` is verified I'll run the downstream tables. 

1. Make sure all addresses that exists in `dim_flipside_<chain>_contracts` exist in the <token>_token_type table
<img width="1145" alt="Screenshot 2024-07-08 at 9 05 54 AM" src="https://github.com/Artemis-xyz/dbt/assets/78228475/11331064-0fd1-4a1d-96cc-51a81521937a">

2. Check to make sure there are no new categories
<img width="1137" alt="Screenshot 2024-07-08 at 9 15 36 AM" src="https://github.com/Artemis-xyz/dbt/assets/78228475/c537adb9-21b6-4598-aedd-3bd5827ef2e7">

3. Check to make sure there are no duplicates
<img width="967" alt="Screenshot 2024-07-08 at 11 04 00 PM" src="https://github.com/Artemis-xyz/dbt/assets/78228475/b567a8c9-cb32-4dcf-91c0-f10e200fa56d">

4. No duplicates in Flipside Contracts
<img width="1060" alt="Screenshot 2024-07-10 at 9 32 59 AM" src="https://github.com/Artemis-xyz/dbt/assets/78228475/91ec6097-01ba-46f8-b692-5d944d5d33cc">

5. number of new contracts
Before adding evm chains `dim_flipside_contracts` had 50224127 entries, now it has 536485150 entries. This is 486,261,023 more contracts added to your flipside contracts table. Our contracts gold table has 55851337 rows where the most tagging comes from solana a chain that relies on flipside tagging. 
<img width="940" alt="Screenshot 2024-07-10 at 9 36 56 AM" src="https://github.com/Artemis-xyz/dbt/assets/78228475/484ef417-14ec-40a6-9e73-10fe9189d79d">

Adding the flipside tagging to your system will make ethereum, polygon and bsc our most tagged chains.
<img width="864" alt="Screenshot 2024-07-10 at 9 39 08 AM" src="https://github.com/Artemis-xyz/dbt/assets/78228475/c2634a8e-0024-4185-9cb7-b26afa77ee4a">


6. New Namespaces
We are going to introduce 1.4K new namespaces into our tagging system. We make sure to not add any of the token namespaces from flipside.
<img width="1134" alt="Screenshot 2024-07-10 at 11 32 45 AM" src="https://github.com/Artemis-xyz/dbt/assets/78228475/bc1305a2-2602-40ff-ba33-05456bd56eb6">

## To Do
1. Adding an new model for EVM chains so I will need to make sure to update `refresh_labels_job` to have `++dim_flipside_contracts`
2. Run new models
3. Unify the new namespaces
